### PR TITLE
Refactor SparseConcat CPU implem and add two tests

### PR DESCRIPTION
**Pull request by benbarsdell**
- This is in preparation for adding a GPU implementation.
- Adds early-exit when the output is empty (no non-zeros).
- Adds tests for all/some inputs being empty (no non-zeros).
- No functional change.

cc @nluehr @sanjoy

### CODE REVIEW

**Comment by benbarsdell** [comment]
Friendly ping.

File: tensorflow/python/kernel_tests/sparse_concat_op_test.py [hunk 1]
```diff
@@ -229,6 +238,45 @@ def testConcat3(self):
...
+      concat_dim = 1
+      sp_concat = sparse_ops.sparse_concat(concat_dim, [sp_a, sp_b, sp_c])
+
+      self.assertEqual(sp_concat.indices.get_shape(), [output_nnz, 3])
+      self.assertEqual(sp_concat.values.get_shape(), [output_nnz])
+      self.assertEqual(sp_concat.dense_shape.get_shape(), [3])
+
+      concat_out = self.evaluate(sp_concat)
+
+      self.assertAllEqual(concat_out.indices, sp_b.indices + [0, 7, 0])
```
Comment (penpornk) [inline tensorflow/python/kernel_tests/sparse_concat_op_test.py:276]: Nit: Should we replace `7` with `sp_a.dense_shape[1]`?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm sorry for the delay! I have one minor comment.

File: tensorflow/python/kernel_tests/sparse_concat_op_test.py (hunk 1 above)
Comment (benbarsdell) [reply tensorflow/python/kernel_tests/sparse_concat_op_test.py]: Done

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Pull request merged**